tests/util/test_distance.py

```diff
@@ -0,0 +1,182 @@
+
+        exceptionThrown = False
+        try:
+            distance_util.meters_to_kilometers('a')
+        except TypeError:
+            exceptionThrown = True
+
+        self.assertTrue(exceptionThrown)
+
+    def test_meters_to_feet(self):
+        """Test conversion from meters to feet."""
+        self.assertEqual(distance_util.meters_to_feet(0), 0)
+        self.assertEqual(distance_util.meters_to_feet(1), 3.28084)
+        self.assertEqual(distance_util.meters_to_feet(500), 1640.42)
+        self.assertEqual(distance_util.meters_to_feet(-1), -3.28084)
+
+        exceptionThrown = False
+        try:
+            distance_util.meters_to_feet('a')
+        except TypeError:
+            exceptionThrown = True
+
+        self.assertTrue(exceptionThrown)
+
+    def test_feet_to_meters(self):
+        """Test conversion from feet to meters."""
+        self.assertEqual(distance_util.feet_to_meters(0), 0)
+        self.assertEqual(distance_util.feet_to_meters(1), 0.3048)
+        self.assertEqual(distance_util.feet_to_meters(500), 152.4)
+        self.assertEqual(distance_util.feet_to_meters(-1), -0.3048)
+
+        exceptionThrown = False
+        try:
+            distance_util.feet_to_meters('a')
+        except TypeError:
+            exceptionThrown = True
+
+        self.assertTrue(exceptionThrown)
+
+    def test_feet_to_miles(self):
+        """Test conversion from feet to miles."""
+        self.assertEqual(distance_util.feet_to_miles(0), 0)
+        self.assertEqual(distance_util.feet_to_miles(1), 0.000189394)
+        self.assertEqual(distance_util.feet_to_miles(500), 0.094697)
+        self.assertEqual(distance_util.feet_to_miles(-1), -0.000189394)
+
+        exceptionThrown = False
+        try:
+            distance_util.feet_to_miles('a')
+        except TypeError:
+            exceptionThrown = True
+
+        self.assertTrue(exceptionThrown)
+
+    def test_miles_to_feet(self):
+        """Test conversion from miles to feet."""
+        self.assertEqual(distance_util.miles_to_feet(0), 0)
+        self.assertEqual(distance_util.miles_to_feet(1), 5280)
+        self.assertEqual(distance_util.miles_to_feet(.5), 2640)
+        self.assertEqual(distance_util.miles_to_feet(-1), -5280)
+
+        exceptionThrown = False
+        try:
+            distance_util.miles_to_feet('a')
+        except TypeError:
+            exceptionThrown = True
+
+        self.assertTrue(exceptionThrown)
+
+    def test_convert_same_unit(self):
+        """Test conversion from any unit to same unit."""
+        self.assertEqual(5, distance_util.convert(5, 'km', 'km'))
+        self.assertEqual(2, distance_util.convert(2, 'm', 'm'))
+        self.assertEqual(10, distance_util.convert(10, 'mi', 'mi'))
+        self.assertEqual(9, distance_util.convert(9, 'ft', 'ft'))
+
+    def test_convert_invalid_unit(self):
+        """Test exception is thrown for invalid units."""
+        exceptionThrown = False
+        try:
```

Comment:
I literally have only used Python for Home-Assistant - so if there are easier ways to do things, such as this - please point me towards them.